src/Nethermind/Nethermind.Init/Steps/EthereumStepsManager.cs

```diff
@@ -201,6 +177,75 @@ private void RunOneRoundOfInitialization(CancellationToken cancellationToken)
             return step;
         }
 
+        private Task RunPluginBeforeStep(IStep step)
+        {
+            Stopwatch stopwatch = Stopwatch.StartNew();
+            return Task.WhenAll(_api.Plugins.Select(p => p.BeforeStep(step.Name).ContinueWith(t =>
+            {
+                if (t.IsFaulted)
+                {
+                    if (_logger.IsError) _logger.Error($"Plugin {p.Name} failed to execute BeforeStep('{step.Name}')", t.Exception);
+                }
+                else if (_logger.IsDebug) _logger.Debug($"Plugin {p.Name} executed BeforeStep('{step.Name}')");
+            })))
+            .ContinueWith(t =>
+            {
+                stopwatch.Stop();
+                if (_logger.IsInfo) _logger.Info($"Plugins executed BeforeStep('{step.Name}') in {stopwatch.ElapsedMilliseconds}ms");
+            });
+        }
+
+        private Task RunPluginAfterStep(IStep step)
+        {
+            Stopwatch stopwatch = Stopwatch.StartNew();
+            return Task.WhenAll(_api.Plugins.Select(p => p.AfterStep(step.Name).ContinueWith(t =>
+            {
+                stopwatch.Stop();
+
+                if (t.IsFaulted)
+                {
+                    if (_logger.IsError) _logger.Error($"Plugin {p.Name} failed to execute AfterStep('{step.Name}')", t.Exception);
+                }
+                else if (_logger.IsDebug) _logger.Debug($"Plugin {p.Name} executed AfterStep('{step.Name}')");
+            })))
+            .ContinueWith(t =>
+            {
+                stopwatch.Stop();
+                if (_logger.IsInfo) _logger.Info($"Plugins executed AfterStep('{step.Name}') in {stopwatch.ElapsedMilliseconds}ms");
+            });
```

Comment:
this code looks redundant in 3 places, not sure if worth to have it single with parameters